File: Java/Digit/program22.java
//Accept two numbers from user and display the power

// Input : 2    4
// Output : 16  (2 * 2 * 2 * 2)

import java.util.*;

class Digit
{
    public int iNo1;
    public int iNo2;

    public Digit(int x, int y)
    {
        iNo1 = x;
        iNo2 = y;
    }

    public void Display()
    {
        System.out.println("The base is: "+this.iNo1);
        System.out.println("The power is: "+this.iNo2);
    }

    public long Power()
    {
        int iCnt = 0;
        long lMult = 1;

        for(iCnt = 1; iCnt <= iNo2; iCnt++)
        {
            lMult = lMult * iNo1;
        }
        return lMult;
    }
}

class program22
{
    public static void main(String arg[])
    {
        int iValue1 = 0;
        int iValue2 = 0;
        long lRet;

        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter the base");
        iValue1 = sobj.nextInt();

        System.out.println("Enter the power");
        iValue2 = sobj.nextInt();

        Digit obj = new Digit(iValue1,iValue2);

        obj.Display();

        lRet = obj.Power();
        System.out.println("Result is : "+lRet);
    }
}

File: C++/Number/program22.cpp
// Accept the number from user and display the addition of that number from 1 to that number.

// Input : 4
// Output : 10  (1+2+3+4)

#include <iostream>
using namespace std;

class Number
{
public:
    int iNo;

    Number(int x)
    {
        iNo = x;
    }

    void Display() 
    {
        cout << "The numbers is: " << this->iNo << endl;
    }

    int Addition()
    {
        int iCnt = 0;
        int iSum = 0;

        if (iNo < 0)
        {
            iNo = -iNo;
        }

        for (iCnt = 1; iCnt <= iNo; iCnt++)
        {
            iSum = iSum + iCnt;
        }
        return iSum;
    }
};

int main()
{
    int iValue = 0;
    int iRet = 0;
    
    cout << "Enter the number" << endl;
    cin >> iValue;

    Number nobj(iValue);

    nobj.Display();

    iRet = nobj.Addition();
    cout << "Addition is: " << iRet << endl;

    return 0;
}
